src/time_travel/patchers/datetime_patcher.py

```diff
@@ -1,30 +1,246 @@
+                    date.day)
+
+
+def datetime_to_fakedatetime(datetime):
+    """Return mocked datetime object from original one."""
+    return FakeDatetime(datetime.year,
+                        datetime.month,
+                        datetime.day,
+                        datetime.hour,
+                        datetime.minute,
+                        datetime.second,
+                        datetime.microsecond,
+                        datetime.tzinfo)
+
+
+class FakeDate(with_metaclass(DateSubclassMeta, 'date', _real_date)):
+    """Mocked datetime.date class."""
+    
+    def __new__(cls, *args, **kwargs):
+        """Return a new mocked date object."""
+        return _real_date.__new__(cls, *args, **kwargs)
+
+    @classmethod
+    def today(cls):
+        """Return today's date."""
+        result = cls._now()
+        return date_to_fakedate(result)
+
+
+FakeDate.min = date_to_fakedate(_real_date.min)
+FakeDate.max = date_to_fakedate(_real_date.max)
+
+
+class FakeDatetime(with_metaclass(DatetimeSubclassMeta, 'datetime',
+                                  _real_datetime, FakeDate)):
+    """Mocked datetime.datetime class."""
+    
+    def __new__(cls, *args, **kwargs):
+        """Return a new mocked datetime object."""
+        return _real_datetime.__new__(cls, *args, **kwargs)
+
+    @classmethod
+    def now(cls, tz=None):
+        """Return a datetime object representing current time."""
+        now = cls._now()
+        if tz:
+            result = tz.fromutc(now.replace(tzinfo=tz)) +\
+                datetime.timedelta(hours=cls._tz_offset())
+        else:
+            result = now
+        return datetime_to_fakedatetime(result)
+
+    @classmethod
+    def today(cls):
+        """Return a datetime object representing current time."""
+        return cls.now(tz=None)
+
+    @classmethod
+    def utcnow(cls):
+        """Return a datetime object representing current time."""
+        result = cls._now()
+        return datetime_to_fakedatetime(result)
+
+
+FakeDatetime.min = datetime_to_fakedatetime(_real_datetime.min)
+FakeDatetime.max = datetime_to_fakedatetime(_real_datetime.max)
+
+
+def pickle_fake_date(datetime_):
+    """Pickle function for FakeDate."""
+    return FakeDate, (
+        datetime_.year,
+        datetime_.month,
+        datetime_.day,
+    )
+
+
+@staticmethod
+def pickle_fake_datetime(datetime_):
+    """Pickle function for FakeDatetime."""
+    return FakeDatetime, (
+        datetime_.year,
+        datetime_.month,
+        datetime_.day,
+        datetime_.hour,
+        datetime_.minute,
+        datetime_.second,
+        datetime_.microsecond,
+        datetime_.tzinfo,
+    )
 
 
 class DatetimePatcher(BasicPatcher):
     """Patcher of the datetime module.
     
     patching:
         - datetime.today
+        - datetime.now
+        - datetime.utcnow
+        - date.today
     """
     
-    def __init__(self, *args, **kwargs):
+    def __init__(self, patched_modules=None, **kwargs):
```

Comment:
I really don't like this interface. The patcher is dynamically loaded (using entry points), right? So what tells the user that he has to pass this argument via `kwargs` for the patcher to work other than the docs? It seems like a really weak spot for the package.

Reply:
Also, what happens if it's None? Nothing is patched?

Reply:
If It's `None`, The code will pass on all modules, and try to find any module using datetime and replace it.
If it's not None, The code will search datetime only in the  given modules

Reply:
Moved it to the basic patcher and opened an issue (#24) to change all patchers to use the patched_modules arg.